src/components/TimerMode.jsx

```diff
@@ -21,14 +19,16 @@ const Timer = ({ duration, status, uid, docid }) => {
     }
   };
 
-  const updateDuration = async (seconds) => {
-    await updateDoc(taskDocRef, {
-      duration: seconds,
-      isActive,
-    });
-  };
-
-  updateDuration(seconds);
+  useEffect(() => {
+    const taskDocRef = doc(db, `users/${uid}/tasks`, `${docid}`);
+    const updateDuration = async (seconds) => {
+      await updateDoc(taskDocRef, {
+        duration: seconds,
+        isActive,
+      });
+    };
+    updateDuration(seconds);
+  }, [uid, docid, isActive, seconds]);
 
   const symbol = isActive ? "◼" : "▶";
```

Comment:
Use directly the value of this instead of putting on a variable since this symbol is only used on 1 place